### PR TITLE
fix(sqlparser): reject reserved words as column name

### LINKED ISSUE #5111: Error reporting is very confusing with extra comma
### Describe the bug

Add an extra comma is a common mistake when the selection list is very long, e.g.:

```sql
create materialized view mv_sum as
select
    count(*) as count_all,
    count(v1) as count_v1,
    count(v2) as count_v2,
    count(v3) as count_v3,
    count(v4) as count_v4,
    sum(v1) as sum_v1,
    sum(v2) as sum_v2,
    sum(v3) as sum_v3,
    sum(v4) as sum_v4,
    min(v1) as min_v1,
    min(v2) as min_v2,
    min(v3) as min_v3,
    min(v4) as min_v4,
    max(v1) as max_v1,
    max(v2) as max_v2,
    max(v3) as max_v3,
    max(v4) as max_v4,
from t;
```

But the error reporting is very confusing:

```
ERROR:  QueryError: Item not found: Invalid column: v1
```

### To Reproduce

```
dev=> select count(v1) as count_v1, from t;
ERROR:  QueryError: Item not found: Invalid column: v1
dev=> select count(v1) as count_v1 from t;
 count_v1
----------
        6
(1 row)
```

### Expected behavior

A better error message

### Additional context

It should be handled in parser, but the error seems occurred in binder.

### PR DESCRIPTION
**Pull request by TennyZhuang**
Signed-off-by: TennyZhuang <zty0826@gmail.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

image

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

Fixes #5111

### CODE REVIEW

**Comment by xiangjinwu** [comment]
Tried the query and it still returned the old confusing error. Looking...

**Comment by xxchan** [comment]
> Tried the query and it still returned the old confusing error. Looking...

Keywords are banned only for table/column alias, but not table/column names..

```sql
dev=> create table table(table int);
CREATE_TABLE
```

**Comment by TennyZhuang** [comment]
😂 maybe we can port another parser framework later...

**Comment by xxchan** [comment]
Just checked the code. It seems to be a feature of `sqlparser-rs` to have keywords as identifiers 😄

https://github.com/risingwavelabs/risingwave/blob/aa827ce2ff2dc0a8c03885701f6c1c7ed7bcc93e/src/sqlparser/src/parser.rs#L2147-L2162

https://github.com/risingwavelabs/risingwave/blob/aa827ce2ff2dc0a8c03885701f6c1c7ed7bcc93e/src/sqlparser/src/parser.rs#L2255-L2261

https://github.com/risingwavelabs/risingwave/blob/aa827ce2ff2dc0a8c03885701f6c1c7ed7bcc93e/src/sqlparser/src/tokenizer.rs#L220-L234

**Comment by xiangjinwu** [comment]
Ready for review

File: src/sqlparser/src/parser.rs [hunk 1]
```diff
@@ -404,6 +404,9 @@ impl Parser {
                 Keyword::ARRAY => Ok(Expr::Array(
                     self.parse_token_wrapped_exprs(&Token::LBracket, &Token::RBracket)?,
                 )),
+                k if keywords::RESERVED_FOR_COLUMN_OR_TABLE_NAME.contains(&k) => {
+                    parser_err!(format!("syntax error at or near \"{w}\""))
+                }
```
Comment (xxchan) [inline src/sqlparser/src/parser.rs:409]: CREATE is not affected

```sql
dev=> create table order(x int, order int);
CREATE_TABLE
dev=> select * from order;
 x | order
---+-------
(0 rows)

dev=> select order.x from order;
ERROR:  QueryError: sql parser error: syntax error at or near "order"
dev=>
```

Comment (xiangjinwu) [reply src/sqlparser/src/parser.rs:409]: Updated to reserve for column in CREATE.

Note that we still allow these words as table names in some cases:
```
dev=> create table order (v1 int);
CREATE_TABLE
dev=> select v1 from order;
 v1 
----
(0 rows)
```

Fixing this requires a bigger effort. As PostgreSQL does allow it when qualified:
```
test=# create table order (v1 int);
ERROR:  syntax error at or near "order"
LINE 1: create table order (v1 int);
                     ^
test=# create table public.order (v1 int);
CREATE TABLE
```

**Pull request merged**